### PR TITLE
Add regenerate function of user's API key

### PR DESCRIPTION
**Pull request by kyoshidajp**
# Description

Add regenerate function of user's API key. I think this feature is useful when an user's API key has leaked out.

Non-admins can regenerate only their own key. Admins can regenerate keys for all user's.

# Screenshot
regenerate_button

regenerate_modal

regenerate_success

regenerate_fail

### CODE REVIEW

File: redash/handlers/users.py [hunk 1]
```diff
@@ -152,6 +152,28 @@ def post(self, user_id):
         }
 
 
+class UserRegenerateApiKeyResource(BaseResource):
+    def post(self, user_id):
+        user = models.User.get_by_id_and_org(user_id, self.current_org)
+        if user.is_disabled:
+            abort(404, message='Not found')
+        if not is_admin_or_owner(user_id):
+            abort(403, message="You cannot regenerate API Key."
```
Comment (arikfr) [inline redash/handlers/users.py]: I think just `abort(403)` is enough. We will not show this UI to non admins once #3234 is resolved.

File: client/app/pages/users/show.js [hunk 2]
```diff
@@ -114,6 +115,33 @@ function UserCtrl(
...
+        .success((data) => {
+          toastr.success('API Key is regenerated.');
+          user.api_key = data.api_key;
+          $scope.disableRegenerateApiKeyButton = false;
+        })
+        .error((response) => {
+          const message =
+            response.message
+              ? response.message
+              : `Cannot regenerate API Key ${response.statusText}`;
```
Comment (arikfr) [inline client/app/pages/users/show.js]:
```suggestion
              : `Failed regenerating API Key: ${response.statusText}`;
```

File: client/app/pages/users/show.js [hunk 3]
```diff
@@ -4,9 +4,10 @@ import { absoluteUrl } from '@/services/utils';
 import template from './show.html';
 import './settings.less';
 
+const regenerateConfirm = { class: 'btn-warning', title: 'Regenerate' };
```
Comment (arikfr) [inline client/app/pages/users/show.js]: Why define it so far from where it's used?

Comment (kyoshidajp) [reply client/app/pages/users/show.js]: I referred to `deleteConfirm` in  https://github.com/getredash/redash/blob/v6.0.0/client/app/pages/data-sources/show.js#L6 as a reference. But `regenerateConfirm` is used only once in `regenerateUserApiKey`. Should I write as an inline?

**Comment by kyoshidajp** [comment]
@arikfr Could you review again?

File: client/app/pages/users/show.js [hunk 4]
```diff
@@ -122,6 +122,34 @@ function UserCtrl(
...
     User.disableUser(user);
   };
+
+  $scope.regenerateUserApiKey = (user) => {
+    const doRegenerate = () => {
+      $scope.disableRegenerateApiKeyButton = true;
+      $http
+        .post(`api/users/${$scope.user.id}/regenerate_api_key`)
+        .success((data) => {
+          toastr.success('API Key is regenerated.');
```
Comment (arikfr) [inline client/app/pages/users/show.js]:
```suggestion
          toastr.success('The API Key has been updated.');
```

File: redash/handlers/users.py [hunk 5]
```diff
@@ -149,6 +149,27 @@ def post(self, user_id):
...
+
+        user.regenerate_api_key()
+        models.db.session.commit()
+
+        self.record_event({
+            'action': 'edit',
+            'object_id': user.id,
+            'object_type': 'user',
+            'updated_fields': ['api_key']
+        })
```
Comment (arikfr) [inline redash/handlers/users.py]: I would use `regnerate_api_key` as `action` and remove the `updated_fields` field.

**Comment by arikfr** [comment]
Thanks!

**Pull request merged**